BSP/Src/BSP_UART.c

```diff
@@ -0,0 +1,85 @@
+#include "BSP_UART.h"
+
+static UART_HandleTypeDef for_interrupts;
+
+//TODO This function needs to be reviewed, and queue creation will likely happen here
```

Comment:
will talk with yall once standard config gets merged but for now this looks decent